src/Client.cpp

```diff
@@ -108,7 +108,9 @@ void UrsaClient::setup_connection() {
         return socket;
     };
 
-    spdlog::info("Connecting to {}", server_addr);
+    if (is_interactive) {
```

Comment:
Consider this: maybe instead of explicitly checking is_interactive, change log level to error in spdlog when we're not interactive?

Reply:
I'm fine with this as a fix though.